armada_jupyter/submissions.py

```diff
@@ -1,78 +1,32 @@
-from typing import List, Optional, Dict, Union
+from typing import List, Optional
 
 import yaml
+from armada_client.armada.submit_pb2 import IngressConfig, ServiceConfig
+from hikaru import load_full_yaml
+from hikaru.model.rel_1_16 import PodSpec
 
-from armada_client.k8s.io.api.core.v1 import generated_pb2 as core_v1
-from armada_client.k8s.io.apimachinery.pkg.api.resource import (
-    generated_pb2 as api_resource,
-)
+from armada_jupyter.constants import PODSPEC_BASE
 
-from armada_jupyter.constants import (
-    YMLSTR,
-    DEFAULT_QUEUE,
-    DEFAULT_PRIORITY,
-    DEFAULT_TIMEOUT,
-    DEFAULT_MEMORY,
-    DEFAULT_CPU,
-)
-
-
-class K8sResourceOptions:
-    """
-    Manages the options for a kubernetes resource request or limit
-    """
 
+class Job:
     def __init__(
         self,
-        cpu: int,
-        memory: str,
-        nvidia_gpu: Union[str, int, None] = None,
-        amd_gpu: Union[str, int, None] = None,
-    ):
-        self.cpu = api_resource.Quantity(string=str(cpu))
-        self.memory = api_resource.Quantity(string=memory)
-        self.nvidia_gpu = nvidia_gpu
-        self.amd_gpu = amd_gpu
-
-        if self.nvidia_gpu and self.amd_gpu:
-            raise ValueError(
-                "Both nvidia_gpu and amd_gpu cannot be set at the same time."
-            )
-
-        # If they exist, they might be a string, so we need to convert them to int
-        if self.nvidia_gpu:
-            self.nvidia_gpu = api_resource.Quantity(string=str(nvidia_gpu))
-
-        if self.amd_gpu:
-            self.amd_gpu = api_resource.Quantity(string=str(amd_gpu))
-
-    def __repr__(self) -> str:
-        return (
-            f"K8sResourceOptions({YMLSTR.cpu}={self.cpu}, "
-            f"{YMLSTR.memory}='{self.memory}', "
-            f"{YMLSTR.nvidia_gpu}={self.nvidia_gpu}, "
-            f"{YMLSTR.amd_gpu}={self.amd_gpu})"
-        )
-
-
-class K8sResources:
-    """
-    Manages the resources for a Kubernetes pod ran by Armada
-    """
-
-    def __init__(
-        self, limits: K8sResourceOptions = None, requests: K8sResourceOptions = None
+        podspec: PodSpec,
+        priority: int = None,
+        namespace: str = None,
+        ingress: List[IngressConfig] = None,
```

Comment:
What is armada-jupyter without a ingress and/or service?  

I think these fields may be required for running these jobs.  They are required in the Job API Spec but we probably need to require them.

Reply:
That is a good points, regardless the typing is wrong here. Should have optional